opennms-dao-api/src/main/java/org/opennms/netmgt/dao/api/InterfaceToNodeCache.java

```diff
@@ -29,13 +29,14 @@
 package org.opennms.netmgt.dao.api;
 
 import java.net.InetAddress;
+import java.util.List;
 import java.util.Optional;
 
 public interface InterfaceToNodeCache {
 
 	void dataSourceSync();
 
-	Iterable<Integer> getNodeId(String location, InetAddress ipAddr);
+	List<Integer> getNodeId(String location, InetAddress ipAddr);
```

Comment:
I am not sure if it is worth sorting the nodeIds. The only usage is in the API layer where a stream of nodeIds is returned. There is no sorting specified for that stream either.

The internally used `Value` class differentiates / sorts by `primaryType` and then `nodeId`. I recognized that this allows for duplicate nodeIds. Therefore I changed the return type to `Set`.

Returning an iterable would work but does not indicate that the collection of node ids has been completely materialized. In addition, it does not indicate that there are no duplicates. IMO iterable as a return type should be used seldomly. For lazy evaluations there is `Stream` and for strict evaluations there is `Collection`. Returning the more specific subtype `Set` instead of `Collection` may be convenient for clients and - in this case - indicates that there are no duplicates.

Reply:
My understanding is, that the sorting of the returned `Iterable` was defined by the order of `Value` which will return nodes which have this interface assigned as primary interface first.

Reply:
Method got removed.